### ISSUE TITLE
Feature request: YubiKey support - HMAC-SHA1 Challenge-Response mode

### ISSUE THREAD

**Issue by seltner** [FrankHausmann/IOSKeePass#10]
I would like to use my yubikey on my iphone via nfc to open my keepass-database-file.

So i could open my keepass-database with my password an my configured yubikey with 
HMAC-SHA1 Challenge-Response mode.

database created with - https://keepassxc.org/docs/#faq-cat-yubikey

https://github.com/Yubico/yubikit-ios

**Comment by FrankHausmann** [comment]
Sorry, at the Moment at make no sense. Before we upgrade to new featured support like YubiKey support
we want to support Version 4 of Keepass DB. 

Another interest issue for me is, 
Why you want a 2FA if KeePassXC and the Database them self not really provide an 2FA

KeePassXC say:
Does KeePassXC support two-factor authentication (2FA) with YubiKeys or OnlyKeys?
    Yes and no. KeePassXC supports YubiKeys for securing a database, but strictly speaking, it's not two-factor authentication. KeePassXC generates a challenge and uses the YubiKey's response to this challenge to enhance the encryption key of your database. So in a sense, it makes your password stronger, but technically it doesn't qualify as a separate second factor, since the expected response doesn't change every time you try to decrypt your database. It does, however, change every time you save your database.

**Comment by seltner** [comment]
Good point. First of all, the new file format should be supported. (this is on the )

And for the other issue. 
Despite the current implementation of KeePassXC  and 2FA with the yubikey.
I want to additionally secure my keepass-database-file with the second factor - 
The standard-possibility is to secure the database with password and as 2FA a file -> key file (Schlüsseldatei)
But cause i have bought me key i wanted to secure the keepass-database-file with the yubikey.
Inspired from: https://curius.de/2021/02/yubikey-teil-iv-keepassxc-absichern/

**Comment by FrankHausmann** [comment]
implementing of KDBX 4 is finished IOSKeePass and ready for Testing.

I have complete changes to another open source library called KeePassKit also find on Github,
Thanks so mutch to the Contributiors of KeePassKit source code

Michael Starke
ChesterLiu
MaximeEpain
Jordan Sechler
but i don´t use there library i use the original source code inside IOSKeePass so that anyone review what is inside.

If you want send me your email address and your nick as first- and surname,so that i can registered you
to AppleStore Connect Testflight group member.

I expected small bugs but i am sure,that 98% will running error free
If you dont want send your data over this comment using fhausmann_at_unicomedv_dot_de

After successfull test i will implement yubikey over nfc

**Comment by seltner** [comment]
hello Frank,
thank you for implementing the database-support for KDBX V4.
i will send you an email with my further contact  details - so we can you within the apple-testflight scenario.

by the way - i have seen the new version 2.0 of the app in the appstore - but unfortunately i can`t update the app.
"Der Entwickler hat diese App aus dem App Store entfernt" !?

**Comment by FrankHausmann** [comment]
Hello Stefan,

yes sure i have it removed, before the shit storm become a hurricane, because the user can login in OLD KDBs
I will insert your contacts to the TestFlight Members

**Comment by seltner** [comment]
Hello Frank,
ah ok, i see.
So you mean, that the new Version 2.0 can`t open the older databases anymore?
Or in other words, the new feature/implementation for database-support KDBX V4 is not backward compatible to older database KDBX versions?
-
Email received - thank you

**Comment by FrankHausmann** [comment]
Maybe yes, look on ISSUE 14#  "can't open database",and also colleagues in US told me today the same.

**Comment by FrankHausmann** [comment]
Hi Stefan,

I will not be supporting Yubikey at the moment, this is because.
1.) There are too big differences in the individual keepass distributions concerning the saving of the secret of the yubikey. Keepass.info stores it via a plugin as a keyfile. Keepassxc has extended the unencrypted header of the DB.

2.) The Apple App Store Review team, needs a video of the whole process flow of what the app does, this video must not be made with the IOS simulator but with a real Apple device. At this point, I drop out because it's too much effort for me. I can understand Apple, because ultimately the NFC controller also plays a big role in Apple Pay.

**Comment by seltner** [comment]
Hi Frank,
i see that it`s a little bit complicated ;)
Maybe you find some time to implement the feature in the future like it is handled in Keepassxc.
No extra plugin is needed - it is natively supported without the need to install extra plugins as for Keepass.info.

And yes, i see that the procedure of creating the video for the Apple App Store Review team is not a task that has to be done in 5 minutes.
They need something like the gif as posted in the blog post here: https://keepassium.com/blog/keepassium-1.10-yubikey/

Thanks you the conversation here - i look forward for new releases :)
